packages/tradershub/src/features/cfd/components/MT5PlatformsList/MT5PlatformsList.tsx

```diff
@@ -48,14 +53,20 @@ const MT5PlatformsList = ({ onMT5PlatformListLoaded }: TMT5PlatformsListProps) =
                 </div>
             )}
             {isFetchedAfterMount &&
-                data?.map(account => {
-                    if (account.is_added)
-                        return <AddedMT5AccountsList account={account} key={`added-mt5-list-${account.loginid}`} />;
+                sortedMt5Accounts?.map(MT5Account => {
+                    if (
+                        MT5Account.is_added &&
+                        MT5Account.is_virtual === activeTradingAccount?.is_virtual &&
+                        MT5Account.account_type === accountType
+                    )
+                        return (
+                            <AddedMT5AccountsList account={MT5Account} key={`added-mt5-list-${MT5Account.loginid}`} />
+                        );
 
                     return (
                         <AvailableMT5AccountsList
-                            account={account as unknown as THooks.MT5AccountsList}
-                            key={`available-mt5-list-${account.market_type}-${account.shortcode}`}
+                            account={MT5Account as unknown as THooks.MT5AccountsList}
```

Comment:
Doesn't `as THooks.MT5AccountsList` type work ? Not sure if unknown is needed here.